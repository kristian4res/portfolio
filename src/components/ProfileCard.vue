<script setup lang="ts">
defineProps<{
  title: string
}>()
</script>

<template>
  <div class="profile-card">
    <h1 class="green">{{ title }}</h1>
    <h3>A DevSecOps software engineer, who's curious about all things tech and food.</h3>
    <div class="flex p-4 justify-evenly">
      <a href="https://www.linkedin.com/in/kristian4res/" target="_blank" rel="noopener"
        >LinkedIn</a
      >
      <a href="https://www.github.com/kristian4res/" target="_blank" rel="noopener">GitHub</a>
      <a
        href="https://a-developers-journal.hashnode.dev/?source=top_nav_blog_home"
        target="_blank"
        rel="noopener"
        >Blog</a
      >
    </div>
  </div>
</template>

<style scoped>
h1 {
  font-weight: 500;
  font-size: 2.6rem;
  position: relative;
  top: -10px;
}

h3 {
  font-size: 1.2rem;
}

.profile-card > h1,
.profile-card > h3 {
  text-align: center;
}

@media (min-width: 1024px) {
  .profile-card > h1,
  .profile-card > h3 {
    text-align: left;
  }
}
</style>
