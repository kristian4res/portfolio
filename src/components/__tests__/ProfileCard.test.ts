import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import ProfileCard from '../ProfileCard.vue'

describe('ProfileCard.vue', () => {
  it('matches snapshot', () => {
    const wrapper = mount(ProfileCard, {
      props: {
        title: 'Kristian Torres',
      },
      slots: {
        default: 'A software engineer that strives to do his best! :)',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
