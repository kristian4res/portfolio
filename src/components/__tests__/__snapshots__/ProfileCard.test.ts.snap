// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileCard.vue > matches snapshot 1`] = `
"<div class=".dark flex flex-col justify-center gap-6">
  <div class="flex flex-col justify-center items-center gap-4">
    <h1 class="w-full font-bold text-4xl text-center lg:text-left">Kristian Torres</h1>
    <h3 class="w-full text-lg break-words text-center lg:text-left">A software engineer that strives to do his best! :)</h3>
  </div>
</div>"
`;
